src/app/common/theme-provider.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  createContext,
+  Dispatch,
+  FC,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+type ComputedTheme = 'dark' | 'light';
+export type UserSelectedTheme = ComputedTheme | 'system';
+
+export const labelMap = new Map<UserSelectedTheme, string>([
+  ['light', 'Light'],
+  ['dark', 'Dark'],
+  ['system', 'System'],
+]);
+
+export const getThemeLabel = (theme: UserSelectedTheme) => {
+  return labelMap.get(theme);
+};
+
+const ThemeContext = createContext<
+  [ComputedTheme, UserSelectedTheme, Dispatch<SetStateAction<UserSelectedTheme>>]
+>([
+  // These values are not used, but are set to satisfy the context's value type.
+  'light',
+  'system',
+  () => {},
+]);
+
+const getSystemTheme = () =>
+  window.matchMedia('prefers-color-scheme: dark').matches ? 'dark' : 'light';
+
+export const ThemeSwitcherProvider: FC = ({ children }) => {
```

Comment:
Recently we've been being explicit with our prop definitions with a `interface ThemeSwitcherProviderProps`. More verbose, but consistent. [We've avoided using `FC` lately](https://medium.com/raccoons-group/why-you-probably-shouldnt-use-react-fc-to-type-your-react-components-37ca1243dd13) as in enforces children when often they're not used (so it's typed wrong)

This is definitely an area where we can align (and refactor), so the whole app is consistent.